### PR TITLE
Feature/refactoring branch jko

### PR DESCRIPTION
**Pull request by SHGODE**
Opening WIP PR for celery and celery flower functionality.  (Running celery through Rest API is still in work in progress.)

### CODE REVIEW

File: app/core/management/utils/xsr_client.py [hunk 1]
```diff
@@ -18,5 +18,5 @@ def read_source_file():
     source_list = [std_source_df]
 
     logger.debug("Sending source data in dataframe format for EVTVL")
-    file_name.delete()
+    # file_name.delete()
```
Comment (jdiouf) [inline app/core/management/utils/xsr_client.py:21]: removing unused code

File: app/openlxp_xia_jko_project/celery.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from celery import Celery
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE',
+                      'openlxp_xia_jko_project.settings')
+
+app = Celery('openlxp_xia_jko_project')
+app.config_from_object('django.conf:settings', namespace='CELERY')
+app.autodiscover_tasks()
+
+# app.conf.beat_schedule = {
```
Comment (jdiouf) [inline app/openlxp_xia_jko_project/celery.py:12]: removing unused code

File: app/openlxp_xia_jko_project/celery.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#     'add-every-30-seconds': {
+#         'task': 'tasks.add',
+#         'schedule': 30.0,
+#         'args': (16, 16)
+#     },
+# }
+
+
+@app.task(bind=True)
+def debug_task(self):
```
Comment (jdiouf) [inline app/openlxp_xia_jko_project/celery.py:32]: Do we still need this?

File: app/openlxp_xia_jko_project/settings.py [hunk 4]
```diff
@@ -136,6 +140,26 @@
...
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
 
+CELERY_BROKER_URL = "redis://redis:6379"
+CELERY_RESULT_BACKEND = "redis://redis:6379"
+CELERY_ACCEPT_CONTENT = ['application/json']
+CELERY_TASK_SERIALIZER = 'json'
+CELERY_RESULT_SERIALIZER = 'json'
+CELERY_TIMEZONE = TIME_ZONE
+#
+# CELERY_BEAT_SCHEDULE = {
```
Comment (jdiouf) [inline app/openlxp_xia_jko_project/settings.py:150]: unused code

File: docker-compose.yml [hunk 5]
```diff
@@ -36,14 +38,67 @@ services:
...
+    ports:
+      - 8888:8888
+    networks:
+      - openlxp
+
 networks:
   openlxp:
     external: true
+
+#volumes:
```
Comment (jdiouf) [inline docker-compose.yml:103]: unsused

File: app/core/management/utils/xsr_client.py (hunk 1 above)
Comment (SHGODE) [reply app/core/management/utils/xsr_client.py:21]: Hi, We need this line, we commented out this code for celery to run-in schedule. We will uncomment this again when we will get API for JKO.

File: app/api/views.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+from core.tasks import xia_workflow
+
+# Create your views here.
+
+
+@api_view(['GET'])
+@permission_classes((permissions.AllowAny,))
+def xia_workflow_api(request):
+    print('XIA workflow api')
```
Comment (mattdrayer) [inline app/api/views.py:14]: Should this `print` instead be a log statement?

File: app/openlxp_xia_jko_project/celery.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    print('Request: {0!r}'.format(self.request))
```
Comment (mattdrayer) [inline app/openlxp_xia_jko_project/celery.py:33]: print -> log?

File: app/openlxp_xia_jko_project/settings.py [hunk 8]
```diff
@@ -136,6 +140,26 @@
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
 
+CELERY_BROKER_URL = "redis://redis:6379"
```
Comment (mattdrayer) [inline app/openlxp_xia_jko_project/settings.py:143]: These values should probably come from environment parameters.

File: app/openlxp_xia_jko_project/settings.py [hunk 9]
```diff
@@ -136,6 +140,26 @@
...
+#     }
+# }
+CACHES = {
+    'default': {
+        'BACKEND': 'django.core.cache.backends.db.DatabaseCache',
+        'LOCATION': 'my_cache_table',
+    }
+}
+
 LOG_PATH = '/etc/debug.log'
```
Comment (mattdrayer) [inline app/openlxp_xia_jko_project/settings.py:163]: This should probably be configured via environment parameters.

**Review by mattdrayer — APPROVED** [review]
Just a few small suggestions, otherwise LGTM.

**Review by jdiouf — APPROVED** [review]

**Pull request merged**